libr/io/io_map.c

```diff
@@ -342,14 +278,14 @@ R_API RIOMap* r_io_map_get_paddr(RIO* io, ut64 paddr) {
 // gets first map where addr fits in
 R_API RIOMap *r_io_map_get(RIO* io, ut64 addr) {
 	r_return_val_if_fail (io, NULL);
-	void **it;
-	r_pvector_foreach_prev (&io->maps, it) {
-		RIOMap *map = *it;
-		if (r_itv_contain (map->itv, addr)) {
-			return map;
-		}
+	const RPVector *skyline = &io->map_skyline;
+	size_t i, len = r_pvector_len (skyline);
+	r_pvector_lower_bound (skyline, addr, i, CMP_END_GTE_PART);
```

Comment:
Looks wrong. I would expect to check `upper_bound(x)-1` which should be the same as `lower_bound(x+1)-1`. Unit tests need to be expanded since this wasn't cought.

Reply:
@kariss couldnt find a way that this fails, so I guess its right